src/treeview/tree-node.component.ts

```diff
@@ -204,11 +206,15 @@ export class TreeNodeComponent implements AfterContentChecked, OnInit, OnDestroy
 	 */
 	nodeClick(event) {
 		if (!this.disabled) {
-			this.selected = true;
-			this.active = true;
 			event.target.parentElement.focus();
-			// Passes event to all nodes to update highlighting & parent to emit
-			this.treeViewService.selectNode({ id: this.id, label: this.label, value: this.value });
+			if (this.selectable || this.children.length === 0) {
+				this.selected = true;
+				this.active = true;
+				// Passes event to all nodes to update highlighting & parent to emit
+				this.treeViewService.selectNode({ id: this.id, label: this.label, value: this.value });
+			} else {
+				this.toggleExpanded(event)
+			}
```

Comment:
```suggestion
			if (this.selectable) {
				this.selected = true;
				this.active = true;
				// Passes event to all nodes to update highlighting & parent to emit
				this.treeViewService.selectNode({ id: this.id, label: this.label, value: this.value });
			} else if(this.children.length > 0) {
				this.toggleExpanded(event)
			}
```

Reply:
This would make the leaf node with `selectable: false` not be actionable...i was thinking more about having this option for expandable nodes as i don't know what could be a valid usecase for leaf nodes not being selectable and being enabled...but if this is a usecase you would want to have...i'm happy to go that way